homeassistant/components/sensor/rest.py

```diff
@@ -0,0 +1,141 @@
+"""
+homeassistant.components.sensor.rest
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The rest sensor will consume JSON responses sent by an exposed REST API.
+
+Configuration:
+
+To use the rest sensor you will need to add something like the following
+to your configuration.yaml file.
+
+sensor:
+  platform: arest
+    name: REST sensor
+    resource: http://IP_ADDRESS/ENDPOINT
+    variable: temperature
+    unit: '°C'
+
+Variables:
+
+name
+*Optional
+The name of the sensor. Default is 'REST Sensor'.
+
+resource
+*Required
+The full URL of the REST service/endpoint that provide the JSON response.
+
+variable
+*Required
+The name of the variable inside the JSON response you want to monitor.
+
+unit
+*Optional
+Defines the units of measurement of the sensor, if any.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.rest.html
+"""
+import logging
+from requests import get, exceptions
+from datetime import timedelta
+
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "REST Sensor"
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Get the REST sensor. """
+
+    resource = config.get('resource', None)
+
+    try:
+        response = get(resource, timeout=10)
```

Comment:
Hmmm. Right now we're doing a mix of everything. I generally prefer to import modules instead of cherry-picking from them (except for constants).

So it's not perse about syntax but what gets imported. This statement for example imports two modules:
```python
from homeassistant.components import sun, group

# equivalent to

import homeassistant.components.sun as sun
import homeassistant.components.group as group
```

When it get to requests, I prefer to `import requests` because `get`, `post` etc is so generic.